index.js

```diff
@@ -1,97 +1,215 @@
+    }
+}
+
+function handleDownStatus(urlName, params, responseDate) {
+    const trailingMentions = buildMentionsForDownStatus(params.mentionsWhenDown);
+    const message = `:warning: Oh no! It looks like ${urlName} (${params.url}) is down.\n${trailingMentions}`;
+    console.log(`${getFormattedDate(responseDate)} - ${message}`);
+
+    if (urlContainer[urlName] === undefined) {
+        send("First check of this session... Endpoint didn't respond :confused:");
+        urlContainer[urlName] = UrlStatus.Pending;
+        return;
+    }
+
+    if (urlErrorContainer[urlName] !== undefined) {
+        restoreLocalErrorAndSend(responseDate, urlName);
+    }
+
+    if (urlContainer[urlName] === UrlStatus.Up) {
+        // First fail! Try again, maybe this was temporary.
+        urlContainer[urlName] = UrlStatus.Pending;
+    } else if (urlContainer[urlName] === UrlStatus.Pending) {
+        // Two fails in a row! Send message, the endpoint is down.
+        urlContainer[urlName] = UrlStatus.Down;
+        send(message);
+    }
+}
+
+function buildMentionsForDownStatus(mentions) {
+    if (mentions === undefined) return '';
+
+    return mentions.reduce((total, mention, index) => {
+        if (mention.user === undefined || mention.group === undefined) return total;
+
+        const leadingText = (index === 0) ? 'cc: ' : ', ';
+        const trailingText = (mention.user) ? `<@${mention.id}>` : `<!subteam^${mention.id}>`;
+
+        return `${total}${leadingText}${trailingText}`
+    }, '');
 }
 
 /* Send message to slack incoming webhook */
 function send(text) {
-    webhook.send(text, function (err, res) {
-        if (err) {
-            console.log('Error:', err);
-        } else {
-            console.log('Message sent: ', res);
+    console.log(webhookUrl);
+    request.post(
+        webhookUrl,
+        {
+            json: { text: text }
+        },
+        function (err, res) {
+            if (err) {
+                console.log(`Error: ${err}`)
+            } else {
+                console.log(`Message sent: ${res}`)
+            }
         }
-    });
+    );
+}
+
+function restoreLocalErrorAndSend(currentDate, urlName) {
+    const localErrorSince = urlErrorContainer[urlName];
+
+    if (localErrorSince === undefined) return;
+
+    const durationInLocalError = moment.duration(moment(currentDate).diff(moment(localErrorSince)))
+    const hasBeenMoreThanOneMinute = durationInLocalError.asSeconds() > 60
+
+    if (hasBeenMoreThanOneMinute) {
+        const formattedDuration = durationInLocalError.format('hh:mm:ss')
+        const message = `Health-check was off for ${formattedDuration}.`;
+        send(message);
+    }
+
+    urlErrorContainer[urlName] = undefined;
 }
 
 /* Get Timestamp */
-function getFormattedDate() {
-    var date = new Date();
+function getFormattedDate(date) {
+    return moment(date).format("DD-MM-YYYY_H:mm:ss")
+}
+
+const parseTimeRe = /^([0-1]\d|2[0-3]):([0-5]\d)$/;
```

Comment:
thinking, we can change this to: `/^([0-1]\d|2[0-3]):([0-5]\d)\s*(|am|pm)$/i` and the nit will return `match[3]` as `''`, `am`, `pm` (and case variants), so this should work:

```js
const parseTimeRe = /^([0-1]\d|2[0-3]):([0-5]\d)\s*(|am|pm)$/i;

function parseTime(str) {
    const match = parseTimeRe.exec(str);
    if (!match) {
        return undefined;
    }
    const offset = match[3].toLowerCase() === 'pm' ? 12 : 0;
    return [parseInt(match[1], 10) + offset, parseInt(match[2], 10)];
}
```